googleapis_auth/lib/auth_io.dart

```diff
@@ -92,7 +92,18 @@ Future<AutoRefreshingAuthClient> clientViaApplicationDefaultCredentials({
     );
   }
 
-  return await clientViaMetadataServer(baseClient: baseClient);
+  try {
+    return await clientViaMetadataServer(baseClient: baseClient);
+  } on Exception catch (err, stack) {
+    Error.throwWithStackTrace(
+      Exception(
```

Comment:
Hrm...I *hate* losing the original exception details. I like this idea, though.

Do we have a standarding "inner error" thing somewhere here?

Reply:
In our case the error will be a "Failed lookup" error. I'm not sure there's a lot of value.
The stacktrace is preserved, so it should point to what originally threw

I'm not sure how we'd better preserve the chain of error. I'm not aware of an official Dart solution to this problem.
In JS we have that `cause` field`, but not in Dart? 🤔